nifi-nar-bundles/nifi-kudu-bundle/nifi-kudu-processors/src/main/java/org/apache/nifi/processors/kudu/AbstractKuduProcessor.java

```diff
@@ -184,10 +200,25 @@ protected KuduClient buildClient(final ProcessContext context) {
         final String masters = context.getProperty(KUDU_MASTERS).evaluateAttributeExpressions().getValue();
         final int operationTimeout = context.getProperty(KUDU_OPERATION_TIMEOUT_MS).evaluateAttributeExpressions().asTimePeriod(TimeUnit.MILLISECONDS).intValue();
         final int adminOperationTimeout = context.getProperty(KUDU_KEEP_ALIVE_PERIOD_TIMEOUT_MS).evaluateAttributeExpressions().asTimePeriod(TimeUnit.MILLISECONDS).intValue();
+        final int workerCount = context.getProperty(WORKER_COUNT).asInteger();
+
+        // Create Executor following approach of Executors.newCachedThreadPool() using worker count as maximum pool size
+        final int corePoolSize = 0;
+        final long threadKeepAliveTime = 60;
+        final Executor nioExecutor = new ThreadPoolExecutor(
+                corePoolSize,
+                workerCount,
+                threadKeepAliveTime,
+                TimeUnit.SECONDS,
+                new SynchronousQueue<>(),
+                new ClientThreadFactory(getIdentifier())
+        );
 
         return new KuduClient.KuduClientBuilder(masters)
                 .defaultOperationTimeoutMs(operationTimeout)
-                .defaultSocketReadTimeoutMs(adminOperationTimeout)
+                .defaultAdminOperationTimeoutMs(adminOperationTimeout)
```

Comment:
This code looks suspect to me it's called adminOperationTimeout but the property is called KUDU_KEEP_ALIVE_PERIOD_TIMEOUT_MS. Not sure what this is and it probably should be cleaned up, but maybe in a seperate change.

Reply:
The `defaultSocketReadTimeoutMs()` setting is deprecated in version 1.14, and the existing variable name `adminOperationTimeout` seemed to be a good match for the `defaultAdminOperationTimeoutMs()` setting.  However, the processor property name and description do not seem to match up as described.  Do you think `defaultAdminOperationTimeoutMs()` should be set using the `operationTimeout` value instead?

Reply:
Reverted the change for now, something that can be addressed separately.